src/header/common/mod.rs

```diff
@@ -264,6 +264,41 @@ macro_rules! header {
             }
         }
     };
+    // Single value cow header
+    ($(#[$a:meta])*($id:ident, $n:expr) => Cow[$value:ty]) => {
+        $(#[$a])*
+        #[derive(Clone, Debug, PartialEq)]
+        pub struct $id(::std::borrow::Cow<'static,$value>);
+        impl $id {
+            /// Creates a new $id
+            pub fn new<I: Into<::std::borrow::Cow<'static,$value>>>(value: I) -> Self {
+                $id(value.into())
+            }
+        }
+        impl ::std::ops::Deref for $id {
+            type Target = ::std::borrow::Cow<'static,$value>;
```

Comment:
So that we don't leak that it is actually a `Cow` inside, let's just `Deref` to the `$value` type instead. This will let us change `Cow` to a `Bytes` or something later, and it should all still just work.